decorations/src/main/java/com/revolut/decorations/DecorationsExt.kt

```diff
@@ -20,9 +20,7 @@ import androidx.recyclerview.widget.RecyclerView
  *
  */
 
-internal fun RecyclerView.forEachViewAdapterPosition(block: (view: View, pos: Int) -> Unit) = (0 until childCount).map { getChildAt(it) }.forEach { view ->
-    getChildAdapterPosition(view).takeUnless { pos -> pos == RecyclerView.NO_POSITION }?.let { pos ->
-        block(view, pos)
-    }
+internal fun RecyclerView.forEachBaseViewHolder(block: (viewHolder: BaseRecyclerViewHolder) -> Unit) = (0 until childCount).map { getChildAt(it) }.forEach { view ->
```

Comment:
add sequence + mapNotNull {it as? BaseRecyclerViewHolder}

Reply:
there would be no profit from Sequence, I'll apply change requested by Igor below, it's more readable that way

Reply:
actually yeah, Sequence.map is better without array instantiation